### ISSUE TITLE
[TF2] Server crashes when an mp_tournament 1 match ends

### ISSUE THREAD

**Issue by ns-phennessy** [Source-Python-Dev-Team/Source.Python#279]
The watchdog process sometimes will say there is a seg fault; however it doesn't always show up.

I unloaded all SP plugins; but it still happend. I unloaded all SM plugins, still happend. I unloaded SP itself, then it _did not_ happen. So looks like it's a problem with SP itself?

I'm not sure what to make of it. Let me know if theres anything I can provide to be able to troubleshoot this issue. I run servers for lots of people so anything I can do would be awesome!

```
Checksum      : c59ab2540567edf8bb6dbec0dc8616c7
Date          : 2019-06-20 02:50:05.342099
OS            : Linux-4.15.0-1034-gcp-x86_64-with-debian-buster-sid
Game          : tf2
SP version    : 690
Github commit : c0e62fa54616e5242f0108cd16a8f2b201fd0b2a
Server plugins:
   00: Metamod:Source 1.10.7-dev
   01: Source.Python, (C) 2012-2019, Source.Python Team.
```

Thanks!

**Comment by Ayuto** [comment]
Can you add ``-debug`` to you command line, so it generates a debug.log?

**Comment by ns-phennessy** [comment]
I figured out the issue. It wasn't actually sourcepython (woot)

I had a plugin that would set the server cfg and change maps. This aparently happend too quickly so it caused SourceTV to not load properly. When it gets to the end of the match, and the auto-recorder is supposed to stop recording, it seg faults because I guess the SourceTV object was a null reference? 

In any case; I added a few delays and now everything works fine haha classic source engine eh?
